app/main.py

```diff
@@ -1 +1,22 @@
-# write your code here
+import shutil
+
+
+def copy_file(command_line: str) -> None:
+    parts = command_line.split()
+
+    if len(parts) == 3 and parts[0] == "cp":
+        source = parts[1]
+        dest = parts[2]
+
+        if source == dest:
+            print("Do nothing")
+            return
+
+        try:
+            with open(source, "r") as file_in, open(dest, "w") as file_out:
+                shutil.copyfileobj(file_in, file_out)
```

Comment:
No need to import additional module, if you can you use `file.write()`
```suggestion
                file_out.write(file_in.read())
```